src/catalog/src/durable/impls/persist.rs

```diff
@@ -809,11 +901,44 @@ impl DurableCatalogState for PersistCatalogState {
 
 /// Extra metadata needed to fence previous catalog.
 #[derive(Debug)]
-pub struct Fence {
+struct Fence {
     /// Previous epoch, if one existed.
     prev_epoch: Option<Epoch>,
 }
 
+/// Initialization status of the catalog.
+#[derive(Debug, Clone)]
+enum InitializedStatus {
+    /// Persist shard is uninitialized. This implies the upper is [`Timestamp::minimum()`].
+    ShardUninitialized,
+    /// Persist shard is initialized, but the catalog is not initialized. This implies the upper is
+    /// [`Timestamp::minimum().step_forward()`]
+    CatalogUninitialized,
+    /// Catalog is initialized with the provided upper.
+    Initialized(Timestamp),
+}
+
+impl InitializedStatus {
+    /// Returns the upper of the catalog shard corresponding to this initialized status.
+    fn upper(self) -> Timestamp {
+        match self {
+            InitializedStatus::ShardUninitialized => Timestamp::minimum(),
+            InitializedStatus::CatalogUninitialized => Timestamp::minimum().step_forward(),
+            InitializedStatus::Initialized(upper) => upper,
```

Comment:
For some reason this feels a little sketchy. Can I rely on `Timestamp::minimum().step_forward()` never changing? Is this even a good idea to tie the state of the catalog to it's upper?